### PR TITLE
Get resource from name

### PR DESCRIPTION
**Pull request by buty4649**
You may want to get a resource from a name.
I think it is more useful to add it to yao.

- - - -
openstackのクライアントは、nameまたUUIDでリソース取得ができる。
それと同じようにyaoでも透過的にリソースの取得ができると便利だと思います。

### CODE REVIEW

File: lib/yao/resources/restfully_accessible.rb [hunk 1]
```diff
@@ -150,5 +156,10 @@ def return_resource(d)
     def return_resources(arr)
       arr.map{|d| return_resource(d) }
     end
+
+    def is_uuid?(str)
+      /^[\da-f]{32}$/ === str or
```
Comment (tnmt) [inline lib/yao/resources/restfully_accessible.rb]: there should be some hosts consist of 32 characters, but I have no idea about ignoring for that case...

File: lib/yao/resources/restfully_accessible.rb [hunk 2]
```diff
@@ -95,13 +95,19 @@ def list_detail(query={})
     def get(id_or_permalink, query={})
       res = if id_or_permalink =~ /^https?:\/\//
               GET(id_or_permalink, query)
-            else
+            elsif is_uuid?(id_or_name_or_permalink)
```
Comment (tnmt) [inline lib/yao/resources/restfully_accessible.rb]: where can we get `id_or_name_or_permalink` ?

Comment (buty4649) [inline lib/yao/resources/restfully_accessible.rb]: oh... 😭

File: lib/yao/resources/restfully_accessible.rb (hunk 1 above)
Comment (buty4649) [inline lib/yao/resources/restfully_accessible.rb]: I have no idea too.Short UUID is almost not used (used in keystone only. maybe). 
So, I will delete it.

File: lib/yao/resources/restfully_accessible.rb [hunk 3]
```diff
@@ -95,13 +95,19 @@ def list_detail(query={})
     def get(id_or_permalink, query={})
       res = if id_or_permalink =~ /^https?:\/\//
               GET(id_or_permalink, query)
-            else
+            elsif is_uuid?(id_or_permalink)
```
Comment (tnmt) [inline lib/yao/resources/restfully_accessible.rb]: name of argument should be particular one that contain name also.

File: lib/yao/resources/restfully_accessible.rb [hunk 4]
```diff
@@ -88,12 +88,16 @@ def get(id_or_permalink, query={})
       res = if id_or_permalink =~ /^https?:\/\//
               GET(id_or_permalink, query)
             else
-              GET([resources_path, id_or_permalink].join("/"), query)
+              find_by_name(id_or_name_or_permalink, query)
```
Comment (tnmt) [inline lib/yao/resources/restfully_accessible.rb]: lost `GET`...

**Comment by buty4649** [comment]
@tnmt review plz !!

File: lib/yao/resources/restfully_accessible.rb [hunk 5]
```diff
@@ -142,5 +148,9 @@ def return_resource(d)
     def return_resources(arr)
       arr.map{|d| return_resource(d) }
     end
+
+    def is_uuid?(str)
```
Comment (takaishi) [inline lib/yao/resources/restfully_accessible.rb]: nits: Method name `uuid?` is more better in Ruby code.

**Comment by buty4649** [comment]
fixed ! review again !

**Comment by buty4649** [comment]
テストを書く 💪

**Comment by buty4649** [comment]
CIが落ちている原因はこれ https://github.com/yaocloud/yao/pull/74/commits/ed2988868cec8a5a215bc56c65cb0b6344626d9b
PR分ければよかった…

**Comment by buty4649** [comment]
@tnmt 
テストコード書いてみました 🙏

File: lib/yao/resources/restfully_accessible.rb [hunk 6]
```diff
@@ -142,5 +148,9 @@ def return_resource(d)
     def return_resources(arr)
       arr.map{|d| return_resource(d) }
     end
+
+    def uuid?(str)
```
Comment (takaishi) [inline lib/yao/resources/restfully_accessible.rb]: 👍

**Comment by buty4649** [comment]
thx!

**Pull request merged**